connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/serializers/GenericRecordTest.java

```diff
@@ -40,17 +44,73 @@
  */
 public class GenericRecordTest {
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testCorrectSerializerInitialization(Boolean useNestedRowsMode) {
+        GenericRecordToRowMutationSerializer serializer = createTestSerializer(useNestedRowsMode);
+
+        assertEquals(TestingUtils.ROW_KEY_FIELD, serializer.getRowKeyField());
+        assertEquals(
+                useNestedRowsMode ? null : TestingUtils.COLUMN_FAMILY,
+                serializer.getColumnFamily());
+        assertEquals(useNestedRowsMode, serializer.getUseNestedRowsMode());
+    }
+
     @Test
-    public void testCorrectRowMutationSerialization() {
-        GenericRecordToRowMutationSerializer serializer = createTestSerializer();
-        RowMutationEntry wantedEntry = TestingUtils.getTestRowMutationEntry();
+    public void testColumnFamilyAndNestedIncompability() {
```

Comment:
These tests could be combined in a parametrized test with multiple booleans, but after checking it, I don't like how it looks like and it makes it harder to read